harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -544,7 +541,8 @@ def _compute_validation_metrics(self) -> workload.Metrics:
             batch_metrics = []
 
             self.validation_loader = cast(torch.utils.data.DataLoader, self.validation_loader)
-            check.gt(len(self.validation_loader), 0)
+            if not self.validation_loader:
+                raise RuntimeError("validation_loader is empty.")
```

Comment:
This is a case where the truthy check is not even kind of a replacement for the length check.

validation_loader will be an DataLoader, which will always be truthy except when it's None.  But it might have `len(validation_loader)==0` even if the validation_loader is not None.

Reply:
Fixed.